spring-pulsar/src/main/java/org/springframework/pulsar/listener/DefaultPulsarMessageListenerContainer.java

```diff
@@ -240,15 +253,172 @@ else if (messageListener != null) {
 						.maxNumMessages(containerProperties.getMaxNumMessages())
 						.maxNumBytes(containerProperties.getMaxNumBytes())
 						.timeout(containerProperties.getBatchTimeoutMillis(), TimeUnit.MILLISECONDS).build();
+
+				Set<String> topicNames = (Set<String>) propertiesToConsumer.remove("topicNames");
+				Map<String, String> properties = (Map<String, String>) propertiesToConsumer.remove("properties");
+
+				ConsumerBuilderCustomizer<T> customizer = builder -> {
+					loadConf(builder, propertiesToConsumer);
+					builder.batchReceivePolicy(batchReceivePolicy);
+				};
 				this.consumer = getPulsarConsumerFactory().createConsumer((Schema) containerProperties.getSchema(),
-						batchReceivePolicy, propertiesToConsumer);
+						topicNames, properties, Collections.singletonList(customizer));
 				Assert.state(this.consumer != null, "Unable to create a consumer");
 			}
 			catch (PulsarClientException e) {
 				DefaultPulsarMessageListenerContainer.this.logger.error(e, () -> "Pulsar client exceptions.");
 			}
 		}
 
+		/*
+		 * Method to work around the issue that ConsumerBuilder::loadConf sets JsonIgnored
+		 * fields of ConsumerConfigurationData to null. To be removed once
+		 * https://github.com/apache/pulsar/issues/11646 is fixed.
```

Comment:
Is this actually the related issue? I know this is the cause of DeadLetterPolicy hackery - maybe they are the same underlying issue?

Reply:
Yes there's also a crash when you have a DLP in the builder. I'll update the comment.

Reply:
done